src/App.tsx

```diff
@@ -1,12 +1,44 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { GoodsList } from './GoodsList';
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+
+  const getAllGoods = async (getFunction: ()=>Promise<Good[]>) => {
+    setGoods(await getFunction());
```

Comment:
better not to do like this, create a separate variable

```suggestion
    const goods = await getFunction();
    setGoods(goods);
```